core/src/main/java/org/apache/calcite/sql/dialect/StarRocksSqlDialect.java

```diff
@@ -109,6 +109,20 @@ public StarRocksSqlDialect(Context context) {
 
   @Override public @Nullable SqlNode getCastSpec(RelDataType type) {
     switch (type.getSqlTypeName()) {
+    case INTEGER:
+      return new SqlDataTypeSpec(
+          new SqlAlienSystemTypeNameSpec(
+              "INT",
+              type.getSqlTypeName(),
+              SqlParserPos.ZERO),
+          SqlParserPos.ZERO);
+    case BIGINT:
+      return new SqlDataTypeSpec(
+          new SqlAlienSystemTypeNameSpec(
+              "BIGINT",
```

Comment:
The BIGINT exists in SqlBasicTypeNameSpec. So maybe we can use :
SqlBasicTypeNameSpec like VARCHAR in MySQL dialect.

Reply:
> The BIGINT exists in SqlBasicTypeNameSpec. So maybe we can use : SqlBasicTypeNameSpec like VARCHAR in MySQL dialect.

Thank you for your suggestion, but I am not very clear about the essential difference between these two writing methods, or is the latter a standardized writing method

Reply:
Hi @fanluoo , AFAIK, If a datatype exists in SqlTypeName, Then we use the SqlBasicTypeNameSpec, meas this datatype is a basic type in Calcite.  SqlAlienSystemTypeNameSpec represents a type name for an alien system. For example, UNSIGNED is a built-in type in MySQL which is synonym of INTEGER. and STRING is synonym of VARCHAR in Spark. UNSIGNED and STRING should used SqlAlienSystemTypeNameSpec.  So BIGINT should used the SqlBasicTypeNameSpec

Reply:
> Hi @fanluoo , AFAIK, If a datatype exists in SqlTypeName, Then we use the SqlBasicTypeNameSpec, meas this datatype is a basic type in Calcite. SqlAlienSystemTypeNameSpec represents a type name for an alien system. For example, UNSIGNED is a built-in type in MySQL which is synonym of INTEGER. and STRING is synonym of VARCHAR in Spark. UNSIGNED and STRING should used SqlAlienSystemTypeNameSpec. So BIGINT should used the SqlBasicTypeNameSpec

@NobiGo The modifications have been made